File: 백준/Silver/1358. 하키/하키.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.StringTokenizer;

public class Main {
    private static int w;
    private static int h;
    private static int x;
    private static int y;

    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st = new StringTokenizer(br.readLine());

        w = Integer.parseInt(st.nextToken());
        h = Integer.parseInt(st.nextToken());
        x = Integer.parseInt(st.nextToken());
        y = Integer.parseInt(st.nextToken());
        int p = Integer.parseInt(st.nextToken());
        int answer = 0;

        for (int i = 0; i < p; i++) {
            st = new StringTokenizer(br.readLine());
            int x1 = Integer.parseInt(st.nextToken());
            int y1 = Integer.parseInt(st.nextToken());
            answer += check(x1, y1);
        }
        System.out.println(answer);
    }

    private static int check(int x1, int y1) {
        if (y1 < y || y1 > y + h) return 0;
        if (x1 >= x && x1 <= x + w) return 1;
        else if (x1 < x && Math.pow(x1 - x, 2) + Math.pow(y1 - (y + (h / 2)), 2) <= Math.pow(h / 2, 2)) return 1;
        else if (x1 > x + w && Math.pow(x1 - (x + w), 2) + Math.pow(y1 - (y + (h / 2)), 2) <= Math.pow(h / 2, 2)) return 1;
        else return 0;
    }
}

File: 백준/Silver/1358. 하키/하키.py
import sys
input = sys.stdin.readline

def solution():
    def check(px, py):
        if py < y or py > y + h: return 0
        if x <= px <= x + w:
            return 1
        elif px < x and (px - x)**2 + (py - (y + (h // 2)))**2 <= (h // 2)**2:
            return 1
        elif px > x + w and  (px - (x + w))**2 + (py - (y + (h // 2)))**2 <= (h // 2)**2:
            return 1
        return 0

    w, h, x, y, p = map(int, input().strip().split())
    answer = 0
    for _ in range(p):
        x1, y1 = map(int, input().strip().split())
        answer += check(x1, y1)
    print(answer)


solution()